syncer/syncer.go

```diff
@@ -1128,23 +1130,30 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		if err != nil {
 			return err
 		}
-
-		// for fresh and all-mode task, flush checkpoint so we could delete metadata file
-		if s.cfg.Mode == config.ModeAll {
-			if err = s.flushCheckPoints(); err != nil {
-				tctx.L().Warn("fail to flush checkpoints when starting task", zap.Error(err))
-			} else if s.cfg.CleanDumpFile {
-				tctx.L().Info("try to remove loaded files")
-				metadataFile := path.Join(s.cfg.Dir, "metadata")
-				if err = os.Remove(metadataFile); err != nil {
-					tctx.L().Warn("error when remove loaded dump file", zap.String("data file", metadataFile), zap.Error(err))
-				}
-				if err = os.Remove(s.cfg.Dir); err != nil {
-					tctx.L().Warn("error when remove loaded dump folder", zap.String("data folder", s.cfg.Dir), zap.Error(err))
-				}
+	}
+	needFlushCheckpoint, err := s.adjustGlobalPointGTID(tctx)
```

Comment:
we could move it to Syncer.Init to avoid resuming triggers it, after `s.checkpoint.Load(tctx)`. And flush checkpoint inside `adjustGlobalPointGTID`

Reply:
( but failpoint should be left here)

Reply:
I don't think we should move this to `Init` because we need to adjust the `Meta.Pos` and it's loaded here. Besides, moving to `Init` may make `syncer.Init` exceed init time.